src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -553,6 +553,10 @@ private String key() {
 
         // super.reload() forces result to be FAILURE, so working around that
         new XmlFile(XSTREAM,new File(getRootDir(),"build.xml")).unmarshal(this);
+        var _execution = execution;
```

Comment:
For what it's worth, my first thought is to instead null out `execution` and reset `executionLoaded` and let them be reloaded by `getExecution` only if needed.

If you prefer to keep the current approach, I would add a `synchronized (this)` block to match `WorkflowRun.getExecution` (unless perhaps typical callers already acquire the monitor for `WorkflowRun` in this case).

Reply:
> null out `execution` and reset `executionLoaded`

:thinking: that may work better. Will check it when I get a moment.

Reply:
It does not work; `getExecution` simply returns null in that case.